packages/transformer/src/test/standalone/IModelTransformerHub.test.ts

```diff
@@ -797,6 +794,71 @@ describe("IModelTransformerHub", () => {
     sinon.restore();
   });
 
+  it("should update aspects when processing changes and detachedAspectProcessing is turned on", async () => {
+    let elementIds: Id64String[] = [];
+    const aspectIds: Id64String[] = [];
+    const sourceIModelId = await createPopulatedIModelHubIModel("TransformerSource", async (sourceSeedDb) => {
+      elementIds = [
+        Subject.insert(sourceSeedDb, IModel.rootSubjectId, "Subject1"),
+        Subject.insert(sourceSeedDb, IModel.rootSubjectId, "Subject2"),
+      ];
+
+      // 10 aspects in total (5 per element)
+      elementIds.forEach((element) => {
+        for (let i = 0; i < 5; ++i) {
+          const aspectProps: ExternalSourceAspectProps = {
+            classFullName: ExternalSourceAspect.classFullName,
+            element: new ElementOwnsExternalSourceAspects(element),
+            identifier: `${i}`,
+            kind: "Document",
+            scope: { id: IModel.rootSubjectId, relClassName: "BisCore:ElementScopesExternalSourceIdentifier" },
+          };
+
+          const aspectId = sourceSeedDb.elements.insertAspect(aspectProps);
+          aspectIds.push(aspectId); // saving for later deletion
+        }
+      });
+
+      return Promise.resolve();
+    });
+
+    const targetIModelId = await createPopulatedIModelHubIModel("TransformerTarget", async () => Promise.resolve());
+
+    try {
+      const sourceDb = await HubWrappers.downloadAndOpenBriefcase({ accessToken, iTwinId, iModelId: sourceIModelId });
+      const targetDb = await HubWrappers.downloadAndOpenBriefcase({ accessToken, iTwinId, iModelId: targetIModelId });
+
+      const transformer = new IModelTransformer(sourceDb, targetDb, {
+        elementAspectExportStrategy: ElementAspectExportStrategy.Detached,
+        includeSourceProvenance: true,
+      });
+
+      // run first transformation
+      await transformer.processChanges({ accessToken });
+      await saveAndPushChanges(targetDb, "First transformation");
+
+      aspectIds.forEach((aspectId) => {
+        sourceDb.elements.deleteAspect(aspectId);
+      });
+
+      await saveAndPushChanges(sourceDb, "Update source");
+
+      await transformer.processChanges({ accessToken, startChangeset: sourceDb.changeset });
+      await saveAndPushChanges(targetDb, "Second transformation");
+
+      const targetElementIds = targetDb.queryEntityIds({ from: Subject.classFullName, where: "EcInstanceId != ?", bindings: [IModel.rootSubjectId] });
+      targetElementIds.forEach((elementId) => {
+        const aspects = targetDb.elements.getAspects(elementId, ExternalSourceAspect.classFullName) as ExternalSourceAspect[];
+        expect(aspects.length).to.be.equal(1); // +1 because provenance aspect was added
+        const aspectAddedAfterFirstTransformation = aspects.find((aspect) => aspect.identifier === "aspectAddedAfterFirstTransformation");
```

Comment:
O_o where does this come from?

Reply:
Somehow forgot to commit part of the test code which inserts additional aspect. The test should be good now.